content/ui-patterns/danger.mdx

```diff
@@ -0,0 +1,72 @@
+
+### Critical data
+
+For critical destructive flows (for example: modifying user access, deleting repositories, making private data public), there are 3 basic treatments you can apply that will ensure the full effects of a change are clear to users and that we minimize loss of trust.
+
+#### Visual treatment
+
+1. Highlight all destructive flows visually. 
+    - All page sections that will trigger a destructive sequence should be highlighted with a [`red border`](https://primer.style/css/utilities/borders#border-colors) and an [`alert` octicon](https://primer.style/octicons/alert-16) to build an accessible, intuitive pattern of recognition. When users notice those details, they should know to tread carefully. Ideally, these will be centralized into a "Danger zone" section.
+    - Any actions that will delete significant user data must use the [`danger button`](https://primer.style/css/components/buttons#danger-button) component.
+
+<img alt='Visual treatment of Danger Zone' src={Img1} />
+
+2. Any actions that have prerequisites should remain disabled until those prerequisites have been completed.
+
+<img alt='Example image of a destructive action whose prerequisites have not been met' src={Img2} />
+
+3. Use standard danger buttons for standard views to avoid drawing unnecessary attention to the danger flows. You want to indicate that they are dangerous without pulling the user's attention away from what their current task.
+
+<img alt='Example image of a danger zone with many "danger" buttons' src={Img3} />
+
+
+#### Secondary confirmation
+
+Take users into a secondary confirmation flow (usually a [box overlay](https://primer.style/css/components/box-overlay)) to ensure they understand the full impact of the decision. 
+
+1. The user should be informed of exactly which data **will** be destroyed and required to confirm that they understand it. This should be explained clearly and concisely in a bulleted list form, using the [`x` octicon](https://primer.style/octicons/x-16) to both draw attention and emphasize that it will be deleted. Don't list out data that will *not* be deleted.
+
+2. Each step in the confirmation flow enables the next step. This forces users to work through each step in sequence and adds a layer of friction that is not prohibitive, but should help minimize errors.
+
+<img alt='Example image of detailed list of data deletions' src={Img4} />
+
+3. The user should be informed of the steps to revert the changes (if reversible) and requested to confirm again.
+
+<img alt='Example image of UI describing if or how data can be recovered' src={Img5} />
+
+4. In cases where the data will delete a user account, an organization, or a repository, the user should be asked to enter their password before confirming.
```

Comment:
@ashygee @vdepizzol following yesterday's conversation, we saw that there _are_ instances where the password is requested currently, and other cases where the name of repo is requested. Should we include that case here too and explain when to do which?